sphinx_gallery/gen_rst.py

```diff
@@ -1000,12 +1000,26 @@ def execute_code_block(
     sys.path.append(os.getcwd())
 
     # Save figures unless there is a `sphinx_gallery_defer_figures` flag
-    match = re.search(
+    defer_figs_match = re.search(
         r"^[\ \t]*#\s*sphinx_gallery_defer_figures[\ \t]*\n?",
         block.content,
         re.MULTILINE,
     )
-    need_save_figures = match is None
+    need_save_figures = defer_figs_match is None
+
+    # Check for `sphinx_gallery_multi_image_block` in this block
+    multi_image_match = re.search(
+        r"^[\ \t]*#\s*sphinx_gallery_multi_image_block[\ \t]*\n?",
+        block.content,
+        re.MULTILINE,
+    )
+    if multi_image_match is not None:
+        script_vars["multi_image"] = py_source_parser.extract_file_config(
+            block.content
+        )["multi_image_block"]
```

Comment:
My attempt at reading the block content and passing the relevant setting to (what becomes) `block_vars`. Looking at this, I feel like there's a more streamlined solution, e.g.:
```
script_vars["multi_image"] = py_source_parser.extract_file_config(block.content).get("multi_image_block")
```